packages/plugin-core/src/commands/GotoNote.ts

```diff
@@ -31,32 +27,20 @@ export { CommandOpts as GotoNoteCommandOpts };
 
 type CommandOutput = { note: NoteProps; pos?: Position } | undefined;
 
-export const findAnchorPos = (opts: { anchor: DNoteAnchor; text: string }) => {
-  const { anchor, text } = opts;
-  let match: Heading | BlockAnchor | undefined = undefined;
-
-  if (anchor.type === "block") {
-    // Anchor leads to a block anchor
-    // TODO: optimize by doing this on startup
-    const blockAnchors = RemarkUtils.findBlockAnchors(text);
-    match = _.find(blockAnchors, (b) => {
-      return b.id == anchor.value;
-    });
-  } else {
-    const anchorSlug = getSlugger().slug(anchor.value);
-    // TODO: optimize by doing this on startup
-    const headers = RemarkUtils.findHeaders(text);
-    match = _.find(headers, (h) => {
-      return getSlugger().slug(h.children[0].value as string) === anchorSlug;
-    });
-  }
+export const findAnchorPos = (opts: {
+  anchor: DNoteAnchor;
+  note: NoteProps;
+}): Position => {
+  const { anchor: findAnchor, note } = opts;
+  let key: string;
+  if (findAnchor.type === "header") key = getSlugger().slug(findAnchor.value);
```

Comment:
we don't currently handle it today but we should be keeping track of anchor values so that we can add the proper suffix for duplicate headers. can we split that out in a separate task and issue?

Reply:
Created #768.